src/pubkey.h

```diff
@@ -305,8 +307,20 @@ struct CExtPubKey {
         return !(a == b);
     }
 
+    friend bool operator<(const CExtPubKey &a, const CExtPubKey &b)
+    {
+        if (a.pubkey < b.pubkey) {
+            return true;
+        } else if (a.pubkey > b.pubkey) {
```

Comment:
I believe that `operator>` implementation is missing for `CPubKey` class:

```cpp
    friend bool operator>(const CPubKey& a, const CPubKey& b)
    {
        return a.vch[0] > b.vch[0] ||
               (a.vch[0] == b.vch[0] && memcmp(a.vch, b.vch, a.size()) > 0);
    }
```

for this line to work and that's why tests fail at the moment.

Reply:
Done. Keep in mind this commit (and the first 15 of this PR) are part of #17034 so comments on them are better suited for that PR.